drivers/sensor/lis2dh/lis2dh.h

```diff
@@ -202,13 +197,11 @@ struct lis2dh_config {
 	int (*bus_init)(const struct device *dev);
 #if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
 	uint16_t i2c_slv_addr;
-#elif DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#endif
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
 	struct spi_config spi_conf;
-#if DT_INST_SPI_DEV_HAS_CS_GPIOS(0)
-	const char *gpio_cs_port;
-	uint8_t cs_gpio;
-	uint8_t cs_gpio_flags;
-#endif /* DT_INST_SPI_DEV_HAS_CS_GPIOS(0) */
+	const char *cs_gpios_label;
 #endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
```

Comment:
It would save a bit of flash to put the I2C and SPI configuration in a union since the device can only be supported on one bus at a time, but it's not a big deal.

Reply:
Yes, correct. I changed topic right now and I do not have time to retest both i2c and spi (it is a must for me the re-testing). But I have in plan to add multi-instance to at least the newer ST sensor we support in zephyr. For those I will for sure take into account your good suggestion! Thanks!

Reply:
Incorporated this change in new push